### PR TITLE
Make QR code scanner working 

### PR DESCRIPTION
**Pull request by mtrabelsi**
This is just an extension to the **window-transport-rack-tracking** branch where the ID are now automatically read from the QR code through the camera.

We can also keep it as a separate branch if we are not going to merge :)

### CODE REVIEW

File: src/CreateNewPage/index.js [hunk 1]
```diff
@@ -41,6 +54,9 @@ class CreateItemPage extends Component {
     return !this.itemId.value;
   };
 
+  handleTextChange = e => {
+
```
Comment (mtrabelsi) [inline src/CreateNewPage/index.js]: this is required when the Component is controlled, we keep it emtpy for now depending for what is required next :)

File: src/CreateNewPage/index.js [hunk 2]
```diff
@@ -108,13 +124,17 @@ class CreateItemPage extends Component {
...
+              value={this.state.id}
+              onChange={this.handleTextChange}
               id="itemId"
               label={`${unit} ID`}
               required
               type="text"
               error={idError}
               errorText="This field is required."
             />
+            <video id="video" width="300" height="200" style={{border: '1px solid gray'}}></video>
```
Comment (mtrabelsi) [inline src/CreateNewPage/index.js]: this is where the camera will be rendered

**Comment by sobolev-alexey** [comment]
This library keeps spinning in a loop a produce exceptions on the console. 
These exceptions can't be turned of, as the library is not providing interface to configure it. 

Here the error is printed out
https://github.com/zxing-js/library/blob/af16fdb2a72b331394dfe22b549703bdfefcae90/src/browser/BrowserCodeReader.ts#L259

Here the `retryIfNotFound` flag set to `true` https://github.com/zxing-js/library/blob/af16fdb2a72b331394dfe22b549703bdfefcae90/src/browser/BrowserCodeReader.ts#L244

Here the private function is called with no chance to modify the flag https://github.com/zxing-js/library/blob/af16fdb2a72b331394dfe22b549703bdfefcae90/src/browser/BrowserCodeReader.ts#L135

In general I'd try to avoid such libraries.
Alternatively I would encourage you to submit PR for them 

image

**Comment by sobolev-alexey** [comment]
One more thing: the camera keeps working after you leave the page. 
This is a big `no-no`

File: src/CreateNewPage/index.js [hunk 3]
```diff
@@ -12,6 +12,7 @@ import { storeItem } from '../store/item/actions';
 import { getFirebaseSnapshot } from '../utils/firebase';
 import { createItemChannel } from '../utils/mam';
 import '../assets/scss/createItemPage.scss';
+import { BrowserQRCodeReader, VideoInputDevice } from '@zxing/library/umd/index.min'
```
[the remark this answers is not in the document]
Comment (mtrabelsi) [reply src/CreateNewPage/index.js]: weird the first time it was not working!

File: package.json [hunk 4]
```diff
@@ -4,6 +4,7 @@
   "private": true,
   "dependencies": {
     "@firebase/app-types": "^0.3.1",
+    "@zxing/library": "^0.7.0",
```
[the remark this answers is not in the document]
Comment (mtrabelsi) [reply package.json]: Actually yarn will pull from package.json, I tried it locally - not sure if this still relevant? what do you think

**Comment by mtrabelsi** [comment]
all the addressed issues has been fixed by the last commit @sobolev-alexey  could you please check it out again? thank you for the deepfeedbacks ;)

File: src/assets/scss/createItemPage.scss [hunk 5]
```diff
@@ -60,7 +67,7 @@
     }
   }
 
-  button.md-btn {
+  button.iota-theme.md-btn {
```
Comment (mtrabelsi) [inline src/assets/scss/createItemPage.scss]: without this class all the global CSS is poluted and impossible to have your own Btn Style, I think we should also use something like BEM we can discuss that later if you want

File: package.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (sobolev-alexey) [reply package.json]: my suggestion actually was to add `yarn.lock` instead of `package-lock.json`

**Pull request merged**